test/unit/org/apache/cassandra/db/compaction/BackgroundCompactionsTest.java

```diff
@@ -440,4 +443,22 @@ public void testSetCompletedNoId()
         BackgroundCompactions backgroundCompactions = new BackgroundCompactions(cfs);
         backgroundCompactions.onCompleted(strategyContainer, null);
     }
+
+    @Test
+    public void periodicReportsTest()
+    {
+        CompactionStrategyOptions options = mock(CompactionStrategyOptions.class);
+        BackgroundCompactions backgroundCompactions = mock(BackgroundCompactions.class);
+        UnifiedCompactionStrategy ucs = mock(UnifiedCompactionStrategy.class);
+        CompactionStrategyStatistics stats = mock(CompactionStrategyStatistics.class);
+
+        when(options.isLogAll()).thenReturn(true);
+        when(options.getLogPeriodMinutes()).thenReturn(1);
+        when(backgroundCompactions.getStatistics(ucs)).thenReturn(stats);
+        when(ucs.getCompactionLogger()).thenReturn(compactionLogger);
+        doCallRealMethod().when(ucs).periodicReport(options, backgroundCompactions);
+
+        ucs.periodicReport(options, backgroundCompactions);
+        Mockito.verify(compactionLogger, times(1)).statistics(eq(ucs), eq("periodic"), any(CompactionStrategyStatistics.class));
+    }
```

Comment:
Since coverage is too low, we will need to also test calling this on `UnifiedCompactionContainer` and `CompactionStrategyManager`.

Maybe the easiest way to do this is to extract the scheduled action in `CompactionManager` to a method, so that we can manually call it, and create a `SizeTieredCompactionStrategy` subclass where we can count calls to `periodicReport` and a table that uses it:
```
Index: test/unit/org/apache/cassandra/db/compaction/CompactionsCQLTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/test/unit/org/apache/cassandra/db/compaction/CompactionsCQLTest.java b/test/unit/org/apache/cassandra/db/compaction/CompactionsCQLTest.java
--- a/test/unit/org/apache/cassandra/db/compaction/CompactionsCQLTest.java	(revision 0f9d47874db250478a3e1ad04ca7ab8f0283c8e2)
+++ b/test/unit/org/apache/cassandra/db/compaction/CompactionsCQLTest.java	(date 1677237098916)
@@ -855,4 +855,14 @@
         if (shouldFind)
             fail("No minor compaction triggered in "+maxWaitTime+"ms");
     }
+
+    @Test
+    public void testPeriodicCompactionsCall()
+    {
+        createTable("CREATE TABLE %s (pk int PRIMARY KEY) WITH compaction = {'class': 'TestCompactionClass'}");
+        TestCompactionClass cs = (TestCompactionClass) getCurrentColumnFamilyStore().getCompactionStrategyContainer().getStrategies().get(0);
+        int prCount = cs.periodicReportsCalled;
+        CompactionManager.periodicReports();
+        assertTrue(cs.periodicReportsCalled > prCount);
+    }
 }
Index: test/unit/org/apache/cassandra/db/compaction/TestCompactionClass.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/test/unit/org/apache/cassandra/db/compaction/TestCompactionClass.java b/test/unit/org/apache/cassandra/db/compaction/TestCompactionClass.java
new file mode 100644
--- /dev/null	(date 1677236939358)
+++ b/test/unit/org/apache/cassandra/db/compaction/TestCompactionClass.java	(date 1677236939358)
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.db.compaction;
+
+import java.util.Map;
+
+public class TestCompactionClass extends SizeTieredCompactionStrategy
+{
+    int periodicReportsCalled = 0;
+
+    public TestCompactionClass(CompactionStrategyFactory factory, Map<String, String> options)
+    {
+        super(factory, options);
+    }
+
+    @Override
+    public void periodicReport(CompactionStrategyOptions testOptions, BackgroundCompactions testBackgroungCompactions)
+    {
+        super.periodicReport(testOptions, testBackgroungCompactions);
+        ++periodicReportsCalled;
+    }
+}
```